staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/monitoring.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversion
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	namespace = "apiserver"
+	subsystem = "crd"
+)
+
+var (
+	latencyBuckets = prometheus.ExponentialBuckets(1, 2, 5)
+)
+
+// converterMonitoringFactory holds metrics for all CRD converters
+type converterMonitoringFactory struct {
+	latencies map[string]*prometheus.HistogramVec
+}
+
+func newConverterMonitoringFactory() *converterMonitoringFactory {
+	return &converterMonitoringFactory{map[string]*prometheus.HistogramVec{}}
+}
+
+var _ crConverterInterface = &converterMonitoring{}
+
+type converterMonitoring struct {
+	delegate  crConverterInterface
+	latencies *prometheus.HistogramVec
+	labels    []string
+	crdName   string
+}
+
+func (c *converterMonitoringFactory) monitor(converterName string, crdName string, converter crConverterInterface) (crConverterInterface, error) {
```

Comment:
what's the purpose of parameterizing this with a converter name?  It's only used one place

Reply:
Right now it is only one place. Does not hurt to keep future converters into account.

Reply:
sure, I'd probably say this is a case of YAGNI, unless you specifically anticipate non-webhook converters.  It's pretty easy to change later.

Reply:
I prefer to keep the codes separated. Having the name "webhook" in the monitoring file does not make sense to me.

Reply:
agree it's odd to put "webhook" inline... a metric creator taking a name param doesn't bother me